crates/pathfinder/src/state/sync.rs

```diff
@@ -905,7 +905,7 @@ async fn l2_update(
         let (transactions_data, events_data): (Vec<_>, Vec<_>) = block
             .transactions
             .into_iter()
-            .zip(block.transaction_receipts.into_iter())
+            .zip(block.transaction_receipts.iter().cloned())
```

Comment:
I think you can instead clone only `events_data`? That way you also don't have to iterate, split and collect it twice.